src/app.ts

```diff
@@ -1,43 +1,76 @@
+import { BigNumberish } from 'ethers';
+
 import {
   ChainName,
-  HyperlaneApp,
-  objMap,
-  promiseObjAll,
+  ChainNameToDomainId,
+  GasRouterApp,
+  RouterContracts,
 } from '@hyperlane-xyz/sdk';
+import { Address } from '@hyperlane-xyz/utils/dist/src/types';
 
 import { HypERC20Contracts, HypERC721Contracts } from './contracts';
+import { TokenRouter } from './types';
 
-export class HypERC20App<Chain extends ChainName> extends HyperlaneApp<
+class HyperlaneTokenApp<
+  Contracts extends RouterContracts<TokenRouter>,
+  Chain extends ChainName,
+> extends GasRouterApp<Contracts, Chain> {
+  async transfer<Origin extends Chain>(
+    origin: Origin,
+    destination: Exclude<Chain, Origin>,
+    recipient: Address,
+    amountOrId: BigNumberish,
+  ) {
+    const originRouter = this.getContracts(origin).router;
+    const destinationDomain = ChainNameToDomainId[destination];
```

Comment:
problematic for PD?

Reply:
Alternative would be to get the chainId from the multiprovider using the new `getChainId` method, but @nambrot mentioned that Hypermint's chainId !== domainId or something?

Reply:
We need to deprecate the use of the global mapping and use a MP-based on instead